waku/v2/protocol/waku_swap/waku_swap.nim

```diff
@@ -1,267 +1,277 @@
-    info "swap handle incoming connection"
-    var message = await conn.readLp(64*1024)
-    # XXX This can be handshake, etc
-    var res = Cheque.init(message)
-    if res.isErr:
-      error "failed to decode rpc"
-      waku_swap_errors.inc(labelValues = [decodeRpcFailure])
-      return
-
-    info "received cheque", value=res.value
-    wakuSwap.handleCheque(res.value)
-
-  proc credit(peerId: PeerId, n: int) {.gcsafe, closure.} =
-    info "Crediting peer: ", peer=peerId, amount=n
-    if wakuSwap.accounting.hasKey(peerId):
-      wakuSwap.accounting[peerId] -= n
-    else:
-      wakuSwap.accounting[peerId] = -n
-    info "Accounting state", accounting = wakuSwap.accounting[peerId]
-    wakuSwap.applyPolicy(peerId)
-
-  # TODO Debit and credit here for Karma asset
-  proc debit(peerId: PeerId, n: int) {.gcsafe, closure.} =
-    info "Debiting peer: ", peer=peerId, amount=n
-    if wakuSwap.accounting.hasKey(peerId):
-      wakuSwap.accounting[peerId] += n
-    else:
-      wakuSwap.accounting[peerId] = n
-    info "Accounting state", accounting = wakuSwap.accounting[peerId]
-    wakuSwap.applyPolicy(peerId)
-    
-  proc applyPolicy(peerId: PeerId) {.gcsafe, closure.} = 
-    # TODO Separate out depending on if policy is soft (accounting only) mock (send cheque but don't cash/verify) hard (actually send funds over testnet)
-
-    #Check if the Disconnect Threshold has been hit. Account Balance nears the disconnectThreshold after a Credit has been done
-    if wakuSwap.accounting[peerId] <= wakuSwap.disconnectThreshold:
-      warn "Disconnect threshhold has been reached: ", threshold=wakuSwap.disconnectThreshold, balance=wakuSwap.accounting[peerId]
-    else:
-      info "Disconnect threshhold not hit"
-
-    #Check if the Payment threshold has been hit. Account Balance nears the paymentThreshold after a Debit has been done
-    if wakuSwap.accounting[peerId] >= wakuSwap.paymentThreshold:
-      warn "Payment threshhold has been reached: ", threshold=wakuSwap.paymentThreshold, balance=wakuSwap.accounting[peerId]
-      #In soft phase we don't send cheques yet
-      #discard wakuSwap.sendCheque()
-    else:
-      info "Payment threshhold not hit"
-
-  wakuSwap.handler = handle
-  wakuSwap.codec = WakuSwapCodec
-  wakuSwap.credit = credit
-  wakuSwap.debit = debit
-  wakuswap.applyPolicy = applyPolicy
-
-# TODO Expression return?
-proc init*(T: type WakuSwap, peerManager: PeerManager, rng: ref BrHmacDrbgContext): T =
-  info "wakuSwap init 2"
-  new result
-  result.rng = rng
-  result.peerManager = peerManager
-  result.accounting = initTable[PeerId, int]()
-  result.text = "test"
-  result.paymentThreshold = 100
-  result.disconnectThreshold = -100
-  result.init()
-
-proc setPeer*(ws: WakuSwap, peer: PeerInfo) =
-  ws.peerManager.addPeer(peer, WakuSwapCodec)
-  waku_swap_peers.inc()
-
-# TODO End to end communication
+## SWAP implements Accounting for Waku. See

```

Comment:
I'm confused by this diff, it looks identical to previous. Maybe formatting issue? Could you make the diff more clear?

Reply:
Yes, make sure that your line endings are correct (should be Unix-like LF characters). See https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/configuring-git-to-handle-line-endings for more.
If you are using VSCode, I highly recommend this extension to render line endings: https://marketplace.visualstudio.com/items?itemName=medo64.render-crlf